### PR TITLE
Define tableName and indexes

### PR DESCRIPTION
**Pull request by nkatzman**
This PR defines attributes in the data-schema that can be used for automated the creation of tables in a datastore

The models that we need to create tables for now all contain tableName and indexes to create

### CODE REVIEW

**Comment by d2lam** [comment]
LGTM 👍

**Comment by FenrirUnbound** [comment]
This is great for defining which index to use for each table. Do we have any future plans for other criteria, like Local Secondary Index in DynamoDB?

File: model/pipeline.js [hunk 1]
```diff
@@ -79,5 +79,21 @@ module.exports = {
...
-    keys: ['scmUrl']
+    keys: ['scmUrl'],
+
+    /**
+     * Tablename to be used in the datastore
+     *
+     * @property tableName
+     * @type {String}
+     */
+    tableName: 'pipelines',
```
Comment (tkyi) [inline model/pipeline.js]: Are we going with `Pipelines` or `pipelines`? (https://github.com/screwdriver-cd/dynamic-dynamodb/pull/1/files#diff-64fbbb5a69337f996786fef9c19e1271R23)

File: model/pipeline.js [hunk 2]
```diff
@@ -79,5 +79,21 @@ module.exports = {
...
+
+    /**
+     * List of indexes to create in the datastore
+     *
+     * @property indexes
+     * @type {Array}
+     */
+    indexes: ['scmUrl']
```
Comment (tkyi) [inline model/pipeline.js]: Is this the `IndexName` or `AttributeName`?

File: model/pipeline.js (hunk 1 above)
Comment (nkatzman) [inline model/pipeline.js]: My vote is for all lowercase plurals i.e. 'pipelines'

File: model/pipeline.js (hunk 2 above)
Comment (nkatzman) [inline model/pipeline.js]: this is meant to be the attributeName to create the global index on

**Comment by nkatzman** [comment]
@FenrirUnbound are there situations where we want a secondary index? Are those things that are useful outside of DynamoDB?

**Comment by stjohnjohnson** [comment]
👍  
mother__have_you_gone_mad

**Comment by jer** [comment]
👍

**Pull request merged**